src/daft-connect/src/op/execute/write.rs

```diff
@@ -54,9 +54,12 @@ impl Session {
                 bail!("Source is required");
             };
 
-            if source != "parquet" {
-                bail!("Unsupported source: {source}; only parquet is supported");
-            }
+            let file_format = match &*source {
+                "parquet" => FileFormat::Parquet,
+                "csv" => FileFormat::Csv,
+                "json" => FileFormat::Json,
```

Comment:
since json isn't implemented, why not bail here instead of inside `data_source`?